### PR TITLE
Warnings cleanup

### PR DESCRIPTION
**Pull request by igchor**
- Create different set of flags for examples and tests.
- Fix warnings
- Enable -Werror

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/33)

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,27 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+if(WIN32)
```
Comment (marcinslusarz) [inline examples/CMakeLists.txt]: This is MSVC-specific, not Windows-specific.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -39,14 +39,50 @@ if(TRACE_TESTS)
...
 if(USE_LLVM_LIBCPP)
 	set(INCLUDE_DIRS ${INCLUDE_DIRS} ${LIBCPP_INCDIR})
 	set(LIBS_DIRS ${LIBS_DIRS} ${LIBCPP_LIBDIR})
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -nostdinc++ -lc++ -lc++abi")
-	# XXX: -lcxxrt instead of -lc++abi on FreeBSD?
+
+	if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+		# -Wno-sign-conversion is necessary to quiet warnings generated by libc++
+		# code which uses p<> template.
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++ -Wno-sign-conversion")
```
Comment (marcinslusarz) [inline tests/CMakeLists.txt]: I don't understand why you replaced other flags...

File: tests/common/unittest.hpp [hunk 3]
```diff
@@ -47,18 +48,27 @@
...
+UT_OUT(const char *format, ...)
 {
-	fprintf(stdout, args...);
+	va_list args_list;
+	va_start(args_list, format);
+	std::vfprintf(stdout, format, args_list);
+	va_end(args_list);
+
 	fprintf(stdout, "\n");
 }
```
Comment (marcinslusarz) [inline tests/common/unittest.hpp]: Why?

File: tests/allocator/allocator.cpp [hunk 4]
```diff
@@ -60,7 +60,7 @@ struct foo {
 	{
 		bar = 1;
 		for (int i = 0; i < TEST_ARR_SIZE; ++i)
-			arr[i] = i;
+			arr[i] = static_cast<char>(i);
```
Comment (pbalcer) [inline tests/allocator/allocator.cpp]: if this were a real code, I would have suggested you change the array type to int, but in test, w/e...

**Comment by marcinslusarz** [comment]
Please rebase this (and possibly others) PR. I want to see Travis jobs succeeds after #39.

**Pull request merged**